### PR TITLE
Adding configuration files 

### PR DESCRIPTION
**Pull request by alexnguyenn**
#120:
Added `.flake8`, `pyproject.toml` (which also contains configuration for black and isort) and `setup.cfg`. 

What still needs to be done:
* Fill out the missing `metadata` fields in `setup.cfg`
* Add `mypy` and `pytest` configurations. 
* Check and correct stuffs (especially dependencies)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+url = https://latticesurgery.com/
+author = George Watkins, Alex Nguyen, Varun Seshardi, Keeland Watkins
+author_email = 
+license = 
+license_file = 
+classifiers =
+    Topic :: Scientific/Engineering
+    Development Status :: 3 - Alpha
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Comment (gwwatkin) [inline setup.cfg:16]: Just curious, have you tried running with 3.8 and 3.7. I've only tried 3.9

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+packages = find:
+python_requires = >=3.7
+install_requires =
+    qiskit>=0.31.0
+    pyzx>=0.6.4
+    igraph>=0.9.8
+    mako>=1.1.5
+
+[options.extras_require]
+dev =
```
Comment (gwwatkin) [inline setup.cfg:37]: Guess `mypy` will go here as well

**Review by gwwatkin — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (alexnguyenn) [reply setup.cfg:16]: Right now I am using 3.9 but I think I used it with 3.7 and 3.8 before. Might need to check again

File: setup.cfg (hunk 2 above)
Comment (alexnguyenn) [reply setup.cfg:37]: Yep, but right now it depends on our progress in [mypy-refactor](https://github.com/latticesurgery-com/lattice-surgery-compiler/tree/mypy-refactor). I don't want to have `mypy` blocking the whole thing.

**Comment by gwwatkin** [comment]
Is this closed to be rebased on the latest branch?

**Pull request closed without merging**